src/order/directed.lean

```diff
@@ -161,6 +162,30 @@ by assumption
 instance order_dual.is_directed_le [has_le α] [is_directed α (≥)] : is_directed αᵒᵈ (≤) :=
 by assumption
 
+section reflexive
+
+lemma directed_on.insert (h : reflexive r) (a : α) {s : set α} (hd : directed_on r s)
+  (ha : ∀ b ∈ s, ∃ c ∈ s, a ≼ c ∧ b ≼ c) : directed_on r (insert a s) :=
+begin
+  rintros x (rfl | hx) y (rfl | hy),
+  { exact ⟨y, set.mem_insert _ _, h _, h _⟩ },
+  { obtain ⟨w, hws, hwr⟩ := ha y hy,
+    exact ⟨w, set.mem_insert_of_mem _ hws, hwr⟩ },
+  { obtain ⟨w, hws, hwr⟩ := ha x hx,
+    exact ⟨w, set.mem_insert_of_mem _ hws, hwr.symm⟩ },
+  { obtain ⟨w, hws, hwr⟩ := hd x hx y hy,
+    exact ⟨w, set.mem_insert_of_mem _ hws, hwr⟩ },
+end
+
+lemma directed_on.singleton (h : reflexive r) (a : α) : directed_on r ({a} : set α) :=
+λ x hx y hy, ⟨x, hx, h _, hx.symm ▸ hy.symm ▸ h _⟩
+
+lemma directed_on_ordered_pair (h : reflexive r) (a b : α) (hab : a ≼ b) :
+  directed_on r ({a, b} : set α) :=
```

Comment:
This is an *unordered* pair.
```suggestion
lemma directed_on_pair (h : reflexive r) {a b : α} (hab : a ≼ b) :
  directed_on r ({a, b} : set α) :=
```

Reply:
I made this mistake to; it's ordered because of `a ≼ b`. Perhaps you're right though that having `ordered` in the name just confuses things.

Reply:
I get that, but "ordered pair" definitely refers to `(a, b)`. To clear up confusion, you should add
```
lemma directed_on_pair' (h : reflexive r) {a b : α} (hab : a ≼ b) :
  directed_on r ({b, a} : set α) :=
```

Reply:
@YaelDillies Done, thanks.